### PR TITLE
chore: upgrade to pnpm v9, regenerate lockfile

### PR DESCRIPTION
**Pull request by AlessioGr**
- regenerates the lockfile
- upgrades pnpm from v8 to v9.7.0 minimum
- ensures playwright does not import payload config. Even after our importmap revamp that made the payload config server-only / node-safe, I was getting these `Error: Invariant: AsyncLocalStorage accessed in runtime where it is not available` errors in combination with pnpm v9 and lockfile regeneration.
This does not happen with pnpm v8, however I'm still blaming playwright for this, as this does not happen in dev and we've had this specific error with playwright in the past when we were importing the payload config. Perhaps it's related to both playwright and the future Next.js process importing the same config file, and not related to the config file containing client-side React code.
Making sure playwright doesn't import the config fixed it (it was importing it through the import map generation). The import map generation is now run in a separate process, and playwright simply waits for it
- One positive thing: this pr fixes a bunch of typescript errors with react-select components. We got those errors because react-select types are not compatible with react 19. lockfile regeneration fixed that (not related to pnpm v9) - probably because we were installing mismatching react versions (I saw both `fb9a90fa48-20240614` and `06d0b89e-20240801` in our lockfile). I have thus removed the caret for react and react-dom in our package.json - now it's consistent

### CODE REVIEW

File: packages/db-mongodb/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "prepublishOnly": "pnpm clean && pnpm turbo build"
   },
   "dependencies": {
+    "@aws-sdk/client-sso-oidc": "^3.614.0",
```
Comment (denolfe) [inline packages/db-mongodb/package.json:36]: Why was this one added? Part of the resolution issue?

File: packages/plugin-relationship-object-ids/package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "payload": "workspace:*"
   },
   "peerDependencies": {
+    "@aws-sdk/client-sso-oidc": "^3.614.0",
```
Comment (denolfe) [inline packages/plugin-relationship-object-ids/package.json:45]: Same question here.

File: test/package.json [hunk 3]
```diff
@@ -68,7 +68,7 @@
     "file-type": "17.1.6",
     "http-status": "1.6.2",
     "jwt-decode": "4.0.0",
-    "lexical": "0.15.0",
+    "lexical": "0.16.1",
```
Comment (denolfe) [inline test/package.json:71]: Any notable changes on this pre 1.0 bump?

File: packages/db-mongodb/package.json (hunk 1 above)
Comment (AlessioGr) [reply packages/db-mongodb/package.json:36]: Sorry, just mentioned the wrong package in the PR description. Changed it to `@aws-sdk/client-sso-oidc`

File: test/package.json (hunk 3 above)
Comment (AlessioGr) [reply test/package.json:71]: This was already bumped from 0.15.0 to 0.16.1 in a previous PR few weeks ago. We just forgot to update that package in the test directory as well

**Comment by AlessioGr** [comment]
Let's wait till pnpm 9.7 for this PR. They recently merged https://github.com/pnpm/pnpm/pull/8330 which I believe might fix this aws peerdependency issue

**Comment by linus-ha** [comment]
pnpm v9.7.0 has just been released  
https://github.com/pnpm/pnpm/releases/tag/v9.7.0

File: packages/plugin-relationship-object-ids/package.json (hunk 2 above)
Comment (AlessioGr) [reply packages/plugin-relationship-object-ids/package.json]: Resolved by pnpm 9.7.0

**Review by denolfe — APPROVED** [review]

**Pull request merged**